### ISSUE TITLE
About RNN4Rec Data

### ISSUE THREAD

**Issue by MLlove0402** [massquantity/LibRecommender#164]
Hi there, i have a question about data. 
Sequential data sometime may have format like: userid, itemid, timestamp, sessionid. How can i train with those format, while its not have label columns

**Comment by massquantity** [comment]
If all of the samples in your data are positive samples, just assign them to one.
```python
>>> data["label"] = 1
```

**Comment by MLlove0402** [comment]
> If all of the samples in your data are positive samples, just assign them to one.
> 
> ```python
> >>> data["label"] = 1
> ```

Thanks you i will try

**Comment by MLlove0402** [comment]
Hi, i just try your suggest. But result is little weird, ndcg and recall score very low
### My code:
import numpy as np
import pandas as pd
from libreco.evaluation import evaluate
from libreco.algorithms import SVD
import torch
import tensorflow as tf
from libreco.data import split_by_ratio_chrono, DatasetPure, random_split
from libreco.training import RNN4RecTrainer


data = pd.read_csv(
    "data/sample_movielens_rating.dat",
    sep="::",
    names=["user", "item", "label", "time"],
)

metrics = [
    "recall",
    "ndcg",
]

tf.compat.v1.reset_default_graph()

train_data, eval_data = random_split(data, multi_ratios=[0.8, 0.2])
train_data, data_info = DatasetPure.build_trainset(train_data, popular_nums=100)
eval_data = DatasetPure.build_evalset(eval_data)

train_data.build_negative_samples(
    data_info, item_gen_mode="random", num_neg=1, seed=2020
)

eval_data.build_negative_samples(
    data_info, item_gen_mode="random", num_neg=1, seed=2222
)

svd = SVD(
    "ranking",
    data_info,
    loss_type="cross_entropy",
    embed_size=16,
    n_epochs=10,
    lr=0.001,
    reg=None,
    batch_size=256,
    num_neg=1,
)
svd.fit(
    train_data,
    verbose=2,
    shuffle=True,
    eval_data=eval_data,
    metrics=metrics,
)

### Result

random neg item sampling elapsed: 0.214s
random neg item sampling elapsed: 0.063s
Training start time: 2023-01-04 09:10:09
train: 100%|██████████| 625/625 [00:03<00:00, 197.65it/s]
Epoch 1 elapsed: 3.280s
	 train_loss: 0.6828
eval_rec: 100%|██████████| 4654/4654 [00:01<00:00, 2693.61it/s]
	 eval recall@10: 0.0370
	 eval ndcg@10: 0.0524
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 389.14it/s]
Epoch 2 elapsed: 1.645s
	 train_loss: 0.6595
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6804.94it/s]
	 eval recall@10: 0.0374
	 eval ndcg@10: 0.0523
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 378.49it/s]
Epoch 3 elapsed: 1.693s
	 train_loss: 0.6378
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6804.86it/s]
	 eval recall@10: 0.0375
	 eval ndcg@10: 0.0510
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 385.98it/s]
Epoch 4 elapsed: 1.658s
	 train_loss: 0.6149
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6710.34it/s]
	 eval recall@10: 0.0371
	 eval ndcg@10: 0.0502
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 379.12it/s]
Epoch 5 elapsed: 1.690s
	 train_loss: 0.5881
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 5909.57it/s]
	 eval recall@10: 0.0374
	 eval ndcg@10: 0.0506
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 388.41it/s]
Epoch 6 elapsed: 1.652s
	 train_loss: 0.5577
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6812.91it/s]
	 eval recall@10: 0.0372
	 eval ndcg@10: 0.0494
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 382.33it/s]
Epoch 7 elapsed: 1.675s
	 train_loss: 0.5263
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6507.24it/s]
	 eval recall@10: 0.0365
	 eval ndcg@10: 0.0491
==============================
train: 100%|██████████| 625/625 [00:01<00:00, 393.52it/s]
Epoch 8 elapsed: 1.627s
	 train_loss: 0.496
eval_rec: 100%|██████████| 4654/4654 [00:00<00:00, 6730.29it/s]
	 eval recall@10: 0.0347
	 eval ndcg@10: 0.0474
	
### environment
numpy==1.23.4
LibRecommender==0.12.0
Python 3.8.16
